### PR TITLE
adding rest api tests and changing project structure

### PR DESCRIPTION
**Pull request by jujga**
* linter's warning about string's length please ignore.
* using linter: in the pycharm's terminal type 'pycodestyle --exclude=venv tests'

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1 @@
-      Автотесты для тестирования добавления в Избранное на сайте https://prom.ua/
-1. Подготовка среды:  в Pycharm через терминал проинсталлировать зависимости: pip install -r requirements.txt
```
Comment (uvidit) [inline README.md:2]: why did you move bullets #1  and  #2 from here. I think it was the best place to keep them here because of the project suppliers will need exactly that things at first - I mean how to set up the project and run tests.
So i suggest to move them back.

File: README.md [hunk 2]
```diff
@@ -1,32 +1 @@
...
-
-   3. test3_add2fav_1_2_1:
-   предустановки: логин и переход в каталог товаров https://prom.ua/Velosipednye-shiny
-   шаги: 
-         кликается по сердцу на товаре по счету номер (счет с нуля) 1
-         кликается по сердцу на товаре по счету номер (счет с нуля) 2
-         кликается по сердцу на товаре по счету номер (счет с нуля) 1
-         проверяется список избранного на странице Избранное на наличие только товара по счету номер 2
-   постусловия: переход на страницу Избранное и очистка списка.
-
```
Comment (uvidit) [inline README.md:31]: it shouldn't be so detailed because of it's extremely hard to support all that staff and keep it up-to-date with only texting in readme file that's not a test management system.
Also  nobody looks to readme after they became familiar with project and can run the test, so I'll bet on that everybody will forget about it, and only newcomers will read it at first .... and probably, there'll find not  up-to date already....
Once again, with increasing tests number it will be hard to support that it will became outdated, so if you'd like to describe something here i's suggest keeping that short, brief and put here only common things those are slowly changing

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+      Автотесты для тестирования добавления в Избранное на сайте https://prom.ua/
+1. Подготовка среды:  в Pycharm через терминал проинсталлировать зависимости: pip install -r requirements.txt
+2. Запуск в терминале Pycharm: "venv\Scripts\pytest.exe -m ui" -тесты IU
```
Comment (uvidit) [inline tests/README.md]: please see comments above

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                               "venv\Scripts\pytest.exe -m rest" -тесты по rest api
```
Comment (uvidit) [inline tests/README.md]: please mark the commands as a code snippets (bordering both sides with ` character), so it would become easier to find them like:

> Подготовка среды: в Pycharm через терминал проинсталлировать зависимости: 

`pip install -r requirements.txt`

File: tests/common.py [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 
 
 def login_steps(main_page):
     main_page.email_button.click()
     main_page.email_field.send_keys(Credentials.login_email)
     main_page.next_button.click()
     main_page.password_field.send_keys(Credentials.login_passw)
-    main_page.comein_button_click()
\ No newline at end of file
+    main_page.comein_button_click()
```
Comment (uvidit) [inline tests/common.py]: > ...comein_button_click()
I'm not sure what did you mean ('common' or 'come in') but i suggest choosing one of these options:
 - ...come_in_button_click()
 - ...common_button_click()

File: tests/test_data/endpoints.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+favorites_url = 'https://my.prom.ua/cabinet/user/favorites'
+velosipednye_shiny_url = 'https://prom.ua/Velosipednye-shiny'
+
```
Comment (uvidit) [inline tests/test_data/endpoints.py]: both url have the common part, so it'd be nice to move it out as a separate part and then use it with string transpolation like:
prom_host = 'https://my.prom.ua/'
fav_url = f'{prom_host}/cabinet/user/favorites'
and etc.
It will save you time if the endpoints number increases or it could become necessary to start testing not only in Prod, but using other environments.

File: tests/test_data/credentials.py [hunk 7]
```diff
@@ -1,3 +1,3 @@
 class Credentials:
     login_email = 'dytaza@mailto.plus'
-    login_passw = 'qwerty654321'
\ No newline at end of file
+    login_passw = 'qwerty654321'
```
Comment (uvidit) [inline tests/test_data/credentials.py]: please don't keep passwords and access keys from real accounts inside code.
The repository is public, so everybody can stole them.
I suggest to pass the sensitive data like environment variables and resolve them on the fly running tests.

File: tests/test_data/endpoints.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+gorest_users_url = 'https://gorest.co.in/public/v2/users'
+access_token = 'access-token=4aa39ba4771752ba989306eb2ef68943963538efeaf4d2feedb7b22ebe3368a7'
```
Comment (uvidit) [inline tests/test_data/endpoints.py]: the same issue with sensitive data as mentioned above.
The github history keep all your changes, so once your password appeared in the code everybody can retrieve it and stole your accounts.

File: tests/test_suite1/conftest.py [hunk 9]
```diff
@@ -0,0 +1,2 @@
+from tests.fixtures.drivers import *
+from tests.fixtures.pages import *
```
Comment (uvidit) [inline tests/test_suite1/conftest.py]: will it work if you move that import lines to __init__.py ?
so you could become more simple.
...i'm not sure it will work, so please check it by your own :)

File: tests/test_suite2/test_rest_api.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            "status": "active"}
+    # как нибудь потом набить коллекцию id записей в глобальную классовую переменную и удалить в тирдауне к сессии
+    requests.delete(gorest_users_url+'/'+str(users_id_2_del), headers=RequestComponents.headers)
+
+
+@pytest.mark.rest
+def test2_get_user():
+    user_id = 4327
+    response = requests.request(
+        "GET", gorest_users_url+'/'+str(user_id), headers=RequestComponents.headers)
```
Comment (uvidit) [inline tests/test_suite2/test_rest_api.py]: please use string interpolation for that like:

`f"{gorest_users_url}/{user_id}"`

File: tests/test_suite2/test_rest_api.py [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+@pytest.mark.rest
+def test3_update_user():
+    user_id = 5099
+    payload = json.dumps({
+        "name": "Changed Name",
+        "email": "changed_mail@15ce.com",
+        "status": "inactive"
+    })
+    response = requests.request(
+        'PATCH', gorest_users_url+'/'+str(user_id), headers=RequestComponents.headers, data=payload)
```
Comment (uvidit) [inline tests/test_suite2/test_rest_api.py]: I saw the exact endpoint before - maybe it could be reused?

File: tests/fixtures/pages.py [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+from tests.fixtures.drivers import driver
+import tests.test_data.endpoints
+from tests import common
+
+
+@pytest.fixture
+def logined_page(driver: driver):
+    driver.get(tests.test_data.endpoints.velosipednye_shiny_url)
+    main_page = MainPage(driver)
+    main_page.comein_link.click()
```
Comment (uvidit) [inline tests/fixtures/pages.py:12]: not clear naming or typo

**Review by uvidit — CHANGES_REQUESTED** [review]
Also i suggest to change project structure.
Now you have suites named using numbers - that's not obvious what they do.
Inside suites there're tests those have similar single type ui or api - and it will be hard to add a such ending every time to the all test inside suite where there're only tests of one type.
So maybe it could be better to put 'ui' or 'api' to the directory level inside suite names and keep only one type tests inside, but with no 'api'/'ui' part in their names?
I mean
 - 'api_suite'
   * rest_test
   * soap_test
   * smoke_test
   etc.
 - 'ui_suite'
   * login_test
   * fav_test
   etc

**Comment by jujga** [comment]
according to Naming conventions of pytest (Pytest file should start with either test_filename or  filename_test

Naming convention for all functions should start with test_)
I leave substr "test" in a begin of each func test name.
Directory's names renamed according to remarks.

**Comment by jujga** [comment]
> Also i suggest to change project structure. Now you have suites named using numbers - that's not obvious what they do. Inside suites there're tests those have similar single type ui or api - and it will be hard to add a such ending every time to the all test inside suite where there're only tests of one type. So maybe it could be better to put 'ui' or 'api' to the directory level inside suite names and keep only one type tests inside, but with no 'api'/'ui' part in their names? I mean
> 
> * 'api_suite'
>   
>   * rest_test
>   * soap_test
>   * smoke_test
>     etc.
> * 'ui_suite'
>   
>   * login_test
>   * fav_test
>     etc

changed accroding to remarks and pytest naming convention

File: README.md (hunk 2 above)
Comment (jujga) [reply README.md:31]: removed tests's description in the tests_descr_draft.txt

File: tests/README.md (hunk 4 above)
Comment (jujga) [reply tests/README.md]: done (не дон! :))) )

File: tests/common.py (hunk 5 above)
Comment (jujga) [reply tests/common.py]: replaced pard of name "comein" -> "submit" (refactored)

File: tests/test_data/endpoints.py (hunk 8 above)
Comment (jujga) [reply tests/test_data/endpoints.py]: It doesn't matter. I don't value this password much )

File: tests/test_suite1/conftest.py (hunk 9 above)
Comment (jujga) [reply tests/test_suite1/conftest.py]: No...It's a special file. It should be named exactly conftest.py. If use import in the conftest.py, all became imported in all files located in the same dir as  conftest.py

File: tests/test_suite2/test_rest_api.py (hunk 10 above)
Comment (jujga) [reply tests/test_suite2/test_rest_api.py]: good idea!
done
removed getting url to helper func `get_url_userid`

File: tests/test_suite2/test_rest_api.py (hunk 11 above)
Comment (jujga) [reply tests/test_suite2/test_rest_api.py]: removed getting url to helper func `get_url_userid`

File: tests/fixtures/pages.py (hunk 12 above)
Comment (jujga) [reply tests/fixtures/pages.py:12]: what?

File: tests/test_data/credentials.py (hunk 7 above)
Comment (jujga) [reply tests/test_data/credentials.py]: It's a temporary account, I don't value them.

File: tests/test_data/endpoints.py (hunk 6 above)
Comment (jujga) [reply tests/test_data/endpoints.py]: Urls are removed to the class PromEndpoints

File: README.md (hunk 1 above)
Comment (jujga) [reply README.md:2]: U refer to branches...#1 and #2. Is it typo?

**Comment by jujga** [comment]
all comments are answered

**Comment by jujga** [comment]
> all comments are answered

all changes are pushed

File: tests/fixtures/pages.py (hunk 12 above)
Comment (uvidit) [reply tests/fixtures/pages.py:12]: it's about the missing underscore char inside many names, like:
'comein_link'  - it has to be 'come_in_link'
or 'get_url_userid'  - it has to be 'get_url_user_id'
or 'gorest_users_url'  -  it has to be 'go_rest_users_url'

File: README.md (hunk 1 above)
Comment (uvidit) [reply README.md:2]: no that was about the first and the second paragraphs of the readme. I didn't mean any commits.

**Review by uvidit — APPROVED** [review]
LGFM with naming issue comments, I mean missing underscore in several names.

File: tests/fixtures/pages.py (hunk 12 above)
Comment (jujga) [reply tests/fixtures/pages.py:12]: It's not critical. Will be refactored as `sign_in_link` in the next version.
